server/auth/types/multiple/multi_auth.ts

```diff
@@ -140,11 +154,20 @@ export class MultipleAuthentication extends AuthenticationType {
     if (reqAuthType && this.authHandlers.has(reqAuthType)) {
       return await this.authHandlers.get(reqAuthType)!.getAdditionalAuthHeader(request);
     } else {
-      return {};
```

Comment:
I wonder if it makes sense to introduce a new abstract method to help identify the auth type from a request to avoid the looping here. I had to go back and trace why the implementation for `getCookie` within multi_auth.ts returned an empty object and its because this block in [authentication_type.ts](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/authentication_type.ts#L115-L134) is really only applicable for JWT and proxy. All other flows rely on calling the `/auth/login` (or similar) API which populates the cookies on response. 

wdyt about something like:

```
protected abstract isRequestFor(
    request: OpenSearchDashboardsRequest,
  ): boolean;
```

which can be used to determine if this request is for the respective auth type so that it can delegate respective calls to the handler.

Reply:
I would say you know better than I do. I don't fully understand how all the login and cookie handling stuff is supposed to work.

Reply:
Are you minded to merge this as-is if the tests pass, or do you want to introduce the above abstraction nevertheless?

Reply:
Can we leave a comment in here and `getCookie` that this logic is only applicable to JWT? I think having passing tests would be good for this PR. 

Ideally, I'd like to have a re-usable method to identify the concrete authtype from the existing cookie or request. If the cookie exists, get it from the cookie, otherwise check to see if the request is for a particular authenticator.

Reply:
Have added comments.